modules/nix/nixpkgs.nix

```
@@ -44,72 +90,260 @@ let
     check = lib.isFunction;
     merge = lib.mergeOneOption;
   };
+
+  # TODO: Remove backwards compatibility hack when dropping
+  # 22.11 support.
+  pkgsType = types.pkgs or (types.uniq types.attrs) // {
+    # This type is only used by itself, so let's elaborate the description a bit
+    # for the purpose of documentation.
+    description = "An evaluation of Nixpkgs; the top level attribute set of packages";
+  };
+
+  hasBuildPlatform = opt.buildPlatform.highestPrio < (mkOptionDefault {}).priority;
+  hasHostPlatform = opt.hostPlatform.isDefined;
+  hasPlatform = hasHostPlatform || hasBuildPlatform;
+
+  # Context for messages
+  hostPlatformLine = optionalString hasHostPlatform "${showOptionWithDefLocs opt.hostPlatform}";
+  buildPlatformLine = optionalString hasBuildPlatform "${showOptionWithDefLocs opt.buildPlatform}";
+
+  legacyOptionsDefined =
+    optional (opt.system.highestPrio < (mkDefault {}).priority) opt.system
+    ;
+
+  defaultPkgs =
+    if opt.hostPlatform.isDefined
+    then
+      let isCross = cfg.buildPlatform != cfg.hostPlatform;
+          systemArgs =
+            if isCross
+            then {
+              localSystem = cfg.buildPlatform;
+              crossSystem = cfg.hostPlatform;
+            }
+            else {
+              localSystem = cfg.hostPlatform;
+            };
+      in
+      import cfg.source ({
+        inherit (cfg) config overlays;
+      } // systemArgs)
+    else
+      import cfg.source {
+        inherit (cfg) config overlays;
+        localSystem = { inherit (cfg) system; };
+      };
+
+  finalPkgs = if opt.pkgs.isDefined then cfg.pkgs.appendOverlays cfg.overlays else defaultPkgs;
+
 in
 
 {
-  options = {
-    nixpkgs.config = mkOption {
+  options.nixpkgs = {
+    pkgs = mkOption {
```

I find that using the full option namespace makes things easier to find, but `options.nixpkgs` is the irrelevant part joined together 😄 

```suggestion
    nixpkgs.pkgs = mkOption {
```

---

This is directly from the upstream NixOS module. I would prefer to avoid any unnecessary divergence, especially cosmetic, to make future synchronization go smoother.

---

Alright makes sense for this module then, but as a general style guide I lean towards the absolute attributes to make searching easier.